src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -400,6 +402,33 @@ public virtual async Task<ConnectionCompleteParams> Connect(ConnectParams connec
             return completeParams;
         }
 
+        private async Task<ConnectionCompleteParams> TryOpenConnectionWithRetry(ConnectionInfo connectionInfo, ConnectParams connectionParams)
+        {
+            int counter = 0;
+            ConnectionCompleteParams response = null;

```

Comment:
You should be able to make this nullable to be stricter about whether values can be null - `ConnectionCompleteParams? response = null`. And then update the return param too. 

(if you're using VS you should see these show up as yellow squigglies)